### PR TITLE
Enhancment of scsi_add_remove test script

### PR DESCRIPTION
**Pull request by Naresh-ibm**
existing test script cannot be run on vfc disk whic does not has pci_address.
and also, script doesn't has verification part after remove or added the scsi
disk. So modified the script to run using wwid os disk and also added the
verification part to it.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[scsi_add_remove_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3363692/scsi_add_remove_job_log.txt)

File: io/disk/scsi_add_remove.py [hunk 1]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
             self.cancel("lsscsi is not installed")
-        self.pci_device = self.params.get("pci_device", default=None)
-        if not self.pci_device:
-            self.cancel("Please provide PCI address for which you \
-                        want to run the test")
+        self.wwids = self.params.get('wwids', default='').split(',')
+        system_wwids = multipath.get_multipath_wwids()
+        for wwid in self.wwids:
+            if wwid not in system_wwids:
+                self.log.info("%s not present in the system" % wwid)
```
[the remark this answers is not in the document]
Comment (narasimhan-v) [reply io/disk/scsi_add_remove.py]: also, need to decide if some wwids are correct, should we run the test for them or not.
If none of them are correct, we can cancel the test.

File: io/disk/scsi_add_remove.py [hunk 2]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
+
+    def GetscsiId(self, path):
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: `get_scsi_id`

File: io/disk/scsi_add_remove.py [hunk 3]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
-                        time.sleep(5)
-                        self.log.info("%s deleted", scsi_num)
-                        self.log.info("adding back %s", scsi_num)
-                        process.run("echo scsi add-single-device %s > \
-                                     /proc/scsi/scsi", scsi_num_seperated)
-                        time.sleep(5)
-                        self.log.info("%s Added back", scsi_num)
+        for wwid in self.wwids:
+            paths = multipath.get_paths(wwid)
+            for path in paths:
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: Based on which wwids are existing, we need to have the loop for that alone.

File: io/disk/scsi_add_remove.py [hunk 4]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
+                process.run("lsscsi")
+                self.log.info("Deleting %s = %s" % (path, scsi_id))
+                genio.write_file("/sys/block/%s/device/delete" % path, "1")
+                time.sleep(5)
+                new_paths = multipath.get_paths(wwid)
+                if path in new_paths:
+                    self.err_paths.append(path)
+                else:
+                    self.log.info("\n%s = %s deleted\n" % (path, scsi_id))
+                self.log.info("\nadding back %s = %s\n" % (path, scsi_id))
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: Consistency in log message format please

File: io/disk/scsi_add_remove.py [hunk 5]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
+        for wwid in self.wwids:
+            paths = multipath.get_paths(wwid)
+            for path in paths:
+                scsi_id = self.GetscsiId(path)
+                process.run("lsscsi")
+                self.log.info("Deleting %s = %s" % (path, scsi_id))
+                genio.write_file("/sys/block/%s/device/delete" % path, "1")
+                time.sleep(5)
+                new_paths = multipath.get_paths(wwid)
+                if path in new_paths:
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: `if path in multipath.get_paths(wwid):`

File: io/disk/scsi_add_remove.py [hunk 6]
```diff
@@ -33,44 +36,64 @@ def setUp(self):
...
+                    self.err_paths.append(path)
+                else:
+                    self.log.info("\n%s = %s deleted\n" % (path, scsi_id))
+                self.log.info("\nadding back %s = %s\n" % (path, scsi_id))
+                part = "scsi add-single-device %s" % scsi_id
+                genio.write_file("/proc/scsi/scsi", part)
+                time.sleep(5)
+                new_paths = multipath.get_paths(wwid)
+                if path not in new_paths:
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: `if path not in multipath.get_paths(wwid):`

File: io/disk/scsi_add_remove.py.data/README.txt [hunk 7]
```diff
@@ -1,7 +1,7 @@
-This Test removes and adds back a scsi device in all the specified PCI domains
+This Test removes and adds back a scsi device in all the specified wwids
 specified in the 'multiplexer' file.
 This test needs to be run as root.
 
 Inputs Needed (in 'multiplexer' file):
 --------------------------------------
-PCI_devices -   PCI Device entry got from 'lspci' command.
+wwids -   wwids can be fetch from 'multipath -ll' command.
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py.data/README.txt]: or lsscsi command.
There is an option to get that. please document that as well.

File: io/disk/scsi_add_remove.py (hunk 4 above)
Comment (Naresh-ibm) [reply io/disk/scsi_add_remove.py]: @narasimhan-v  you mean here to remove the '\n'?

Comment (narasimhan-v) [reply io/disk/scsi_add_remove.py]: Yes. Either have in all the messages, or remove in all.

File: io/disk/scsi_add_remove.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Naresh-ibm) [reply io/disk/scsi_add_remove.py]: I feel cancel the test even if some of them are not exist.
ultimately the goal is not achieved and waste of time for running only on few paths.

So adding 'self.cancel' here itself.

[the remark this answers is not in the document]
Comment (narasimhan-v) [reply io/disk/scsi_add_remove.py]: okay

**Comment by Naresh-ibm** [comment]
closing this PR in favour of #1188

**Pull request closed without merging**